templates/ec2/sc-ec2-linux-docker-notebook.yaml

```diff
@@ -0,0 +1,491 @@
+                  'Fn::Sub': ${AWS::Region}
+    Properties:
+      ImageId: "ami-0b710e028cd5ff838" # https://github.com/Sage-Bionetworks-IT/packer-base-ubuntu-jammy/tree/v1.0.0
+      InstanceType: !Ref 'EC2InstanceType'
+      SubnetId: !ImportValue
+          'Fn::FindInMap': [AccountToImportParams, !Ref "AWS::AccountId", PrivateSubnet]
+      SecurityGroupIds:
+        - !ImportValue
+          'Fn::FindInMap': [AccountToImportParams, !Ref "AWS::AccountId", VpnSecurityGroup]
+        - !Ref NotebookConnectSecurityGroup
+        - !Ref TgwHubSecurityGroup
+      KeyName: 'scipool'
+      BlockDeviceMappings:
+        -
+          DeviceName: "/dev/sda1"
+          Ebs:
+            DeleteOnTermination: true
+            VolumeSize: !Ref VolumeSize
+            Encrypted: true
+      IamInstanceProfile: !Ref 'InstanceProfile'
+      UserData:
+        Fn::Base64: !Sub
+          -  |
+            #!/bin/bash
+            cfn-init -v --stack ${AWS::StackName} --resource LinuxInstance --configsets ${ConfigSet} --region ${AWS::Region}
+            cfn-signal -e $? --stack ${AWS::StackName} --resource LinuxInstance --region ${AWS::Region}
+          - ConfigSet: !FindInMap [NotebookTypeToInitParams, !Ref NotebookType, ConfigSet]
+      PropagateTagsToVolumeOnCreation: true
+      Tags:
+        - Key: Name
+          Value: !Ref 'AWS::StackName'
+        - Key: Description
+          Value: !Sub "Service Catalog instance created by ${AWS::StackName}"
+        - Key: "ManagedInstanceMaintenanceTarget"
+          Value: "yes"
+        - Key: "PatchGroup"
+          Value: "prod-default"
+    CreationPolicy:
+      ResourceSignal:
+        Timeout: PT20M
+
+  EC2TargetGroup:
+    Type: AWS::ElasticLoadBalancingV2::TargetGroup
+    Metadata:
+      cfn-lint:
+        config:
+          ignore_checks:
+            - W1011
+    Properties:
+      HealthCheckIntervalSeconds: 30
+      HealthCheckProtocol: HTTPS
+      HealthCheckTimeoutSeconds: 15
+      HealthyThresholdCount: 5
+      Matcher:
+        HttpCode: '200'
+      Port: 443
+      Protocol: HTTPS
+      TargetGroupAttributes:
+      - Key: deregistration_delay.timeout_seconds
+        Value: '20'
+      Targets:
+      - Id: !Ref LinuxInstance
+        Port: 443
+      UnhealthyThresholdCount: 3
+      VpcId: !ImportValue
+          'Fn::FindInMap': [AccountToImportParams, !Ref "AWS::AccountId", VPCId]
+      Tags:
+      - Key: Name
+        Value: !Sub 'TargetGroup-${LinuxInstance}'
+
+  AlbListenerRule:
+    Type: Custom::ALBListenerRule
+    Properties:
+      ServiceToken: !ImportValue
+        'Fn::Sub': '${AWS::Region}-cfn-cr-alb-rule-FunctionArn'
+      InstanceId: !Ref LinuxInstance
+      TargetGroupArn: !Ref EC2TargetGroup
+      ListenerArn: !ImportValue
+        'Fn::Sub': '${AWS::Region}-alb-notebook-access-ALBListenerARN'
+  TagInstance:
+    DependsOn: "InstanceProfile"
+    Type: Custom::SynapseTagger
+    Properties:
+      ServiceToken: !ImportValue
```

Comment:
This lambda will take all of the [ENV VARs defined in make_env_vars_file.sh file](https://github.com/Sage-Bionetworks/service-catalog-library/blob/master/templates/ec2/sc-ec2-linux-jumpcloud-notebook.yaml#L285-L298) and apply them as EC2 instance tags.  Some of those tags are very important to how the service catalog works.  For example the value for the `ACCESS_APPROVED_ROLEID` tag is what is used to restrict instance access to the user who provisioned it.  Since the make_env_vars_file.sh is no longer setup in this file i'm going to assume that those tags are no longer getting set.  I think we'll need `make_env_vars_file.sh` or find an alternative way to set those tags.

Reply:
I see. Will add to the config.